### DISCUSSION TITLE
Using framework provided web servers (e.g. spring web) for dashboard

### DISCUSSION THREAD

**Discussion by angry-cellophane** [jobrunr/jobrunr#276] [Ideas]
**Is your feature request related to a problem? Please describe.**
JobRunr has its dashboard http server (api, sse, static content). One one hand it makes it portable and easy to use with many app frameworks like spring, micronaut, etc. One the other hand this approach doesn't allow JobRunr benefit from many features offered by those frameworks like security, standard endpoints (e.g [spring actuator](https://spring.io/guides/gs/actuator-service/) and [prometheus](https://docs.spring.io/spring-metrics/docs/current/public/prometheus)).

Imagine, you're a team that has 20+ spring boot apps, they are all standardized: pipelines (including healthchecks and post-deployment checks), metrics, security through some common libs and configuration.
Even though jobrunr has integration with spring the dashboard doesn't really. It's a snowflake that requires some additional work due to the embedded server, and not everything essential (like security) is available out of the box (but available in spring).  

**Describe the solution you'd like**
No real solution but want to start a discussion on the general direction of jobrunr, where it goes with the dashboard: is the plan to always have an embedded web server, to have another dahsboard app or to have tighter integration with frameworks to leverage their capabilities.

I spent some time noodling with jobrunr and wrote a prototype that uses spring webflux in dashboard - [https://github.com/angry-cellophane/jobrunr-spring-webflux](https://github.com/angry-cellophane/jobrunr-spring-webflux)
It's not ideal, some things are missing but it gives the idea how much work it will require.

**Comment by rdehuyss** [comment]
To be honest, I want to keep JobRunr agnostic for different frameworks (being Spring, Micronaut or Quarkus).

Adding support for all the different frameworks is easy but maintaining them is hard (upgrades, ... ). If you want to have something that integrates well with Spring, feel free to write your own and use the `StorageProvider` to query the data.
